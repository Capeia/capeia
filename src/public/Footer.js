// @flow
import React, { } from 'react'
import Link from 'found/lib/Link'
import { Grid, Row, Col } from 'react-bootstrap'
import withStyles from 'isomorphic-style-loader/lib/withStyles'
import { generateShareIcon } from 'react-share'

import appConfig from 'config-app'
import s from './Footer.scss'

const TwitterIcon = generateShareIcon('twitter')

const Footer = () => {
  const sections = Object.keys(appConfig.sections).map(slug => (
    <li key={slug}>
      <Link to={`/${slug}`}>{appConfig.sections[slug].name}</Link>
    </li>
  ))

  return (
    <div className={s['page-footer']}>
      {/* TODO: Use sticky arrow next to article instead? */}
      <div className={s.backToTop}>
        <a href='#'>
          <div className={s.arrow} />
          Back to Top
        </a>
      </div>
      <div className={s['footer-logo']}>
        <Link to='/' />
      </div>
      <Grid fluid>
        <Row>
          <Col sm={3} xs={6}>
            <h1>Capeia</h1>
            <ul>
              <li><Link to='/welcome'>About the Platform</Link></li>
              <li><Link to='/faq'>Frequently Asked Questions</Link></li>
              <li><Link to='/info-for-institutes'>Information for Research Institutes</Link></li>
              <li>
                <Link to='https://www.twitter.com/CapeiaScience' target='_blank'>
                  <div className={s.twitterIcon}>
                    <TwitterIcon
                      size={20}
                      round
                      logoFillColor='#2f3240'
                      iconBgStyle={{ fill: 'white', fillOpacity: 0.8 }}
                    />
                  </div>
                  Stay in Touch
                </Link>
              </li>
            </ul>
          </Col>
          <Col sm={3} xs={6}>
            <h1>Ressorts</h1>
            <ul>
              {sections}
            </ul>
          </Col>
          <Col sm={3} xs={6}>
            <h1>Imprint</h1>
            <ul>
              <li><Link to='/who'>Who</Link></li>
              <li><Link to='/terms-of-use'>Terms of Use</Link></li>
              <li><Link to='/contact'>Contact</Link></li>
              {/* TODO: Add dedicated press page */}
              {/* <li><Link to='/press'>Press</Link></li> */}
              <li><Link to='https://www.newswire.com/news/launch-of-the-science-platform-capeia-19265208' target='_blank'>Press</Link></li>
              <li><Link to='/rules-and-guidelines'>Rules & Guidelines</Link></li>
            </ul>
          </Col>
          <Col sm={3} xs={6}>
            <h1>Contribute</h1>
            <ul>
              <li><Link to='/register'>Become an Author</Link></li>
              <li>&nbsp;</li>
              <li>&nbsp;</li>
              <li>&nbsp;</li>
              <li>&nbsp;</li>
              <li><Link to='/a-gap-in-society'>A Gap in Society</Link></li>
            </ul>
          </Col>
        </Row>
        <Row>
          <div className={s.netidee}>
            Part of Netidee call 10
          </div>
        </Row>
      </Grid>
    </div>
  )
}

export default withStyles(s)(Footer)
