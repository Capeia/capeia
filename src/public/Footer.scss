@import '~theme/variables';
@import '~theme/typography';

@mixin footer-fonts {
  @include sans-serif-normal;
  color: darken($text-light, 20);
}

// scss-lint:disable SelectorFormat TODO
.page-footer {
  @include footer-fonts;
  background: $dark;
  padding: 3em 1em;
  position: relative;
  z-index: 10;

  @media (min-width: $screen-sm-min) {
    padding: 3em 6em;
  }

  h1 {
    @include footer-fonts;
    cursor: default;
    font-size: 1.2em;
    margin-bottom: 1em;
    text-transform: uppercase;
    user-select: none;
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  li {
    margin: .5em 0;
  }

  span {
    @include footer-fonts;
    font-size: .9em;
  }

  a {
    @include footer-fonts;
    font-size: .9em;
    transition: color 300ms ease-in-out;

    &:hover {
      color: $text-light;
    }

    &:hover,
    &:active,
    &:focus {
      text-decoration: none;
    }
  }
}

.soon {
  // scss-lint:disable ImportantRule FIXME
  color: lighten($dark, 20) !important;
  cursor: default;
  user-select: none;
}

.backToTop {
  padding-bottom: 2em;
  text-align: center;

  @media (min-width: $screen-md-min) {
    display: none;
  }

  .arrow {
    width: 0;
    height: 0;
    margin: 0 auto;
    border-color: transparent transparent $text-light;
    border-style: solid;
    border-width: 0 1.5em 1.5em;
    opacity: .8;
  }
}

.footer-logo {
  // scss-lint:disable ImportantRule FIXME
  background: url('../../static/capeia_logo_1c_white.svg') center center no-repeat transparent;
  background-size: contain !important;
  height: 4em;
  margin: 0 auto 2em;
  opacity: .75;
  width: 15em;

  a {
    display: block;
    height: 100%;
    width: 100%;
  }
}

.twitterIcon {
  display: inline-block;
  margin-right: .4em;
  vertical-align: text-bottom;
}

.netidee {
  padding-top: 2em;
  font-size: .6em;
  text-align: center;
  text-transform: uppercase;
  opacity: 0.7;

  &::before {
    display: block;
    padding: 1.4em;
    margin-bottom: 1em;
    content: '';
    background: url('../../static/netidee_logo_1c.png') center center no-repeat;
    background-size: contain;
  }
}

@media print {
  .page-footer {
    display: none;
  }
}
