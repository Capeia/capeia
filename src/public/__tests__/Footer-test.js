// @flow
/* eslint-env jest, jasmine */

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Footer from '../Footer'
import appConfig from 'config-app'

describe('<Footer>', () => {
  it('renders correctly', () => {
    jest.deepUnmock('../Footer')
    const Component = require('../Footer').default
    const tree = renderer.create(<Component />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render a link for every section', () => {
    const wrapper = shallow(<Footer />)
    Object.keys(appConfig.sections).forEach(slug => {
      expect(wrapper.find({ to: `/${slug}`, children: appConfig.sections[slug].name }).length).toEqual(1)
    })
  })
})
