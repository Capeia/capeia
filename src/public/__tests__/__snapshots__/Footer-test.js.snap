// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> renders correctly 1`] = `
<div
  className="page-footer"
>
  <div
    className="backToTop"
  >
    <a
      href="#"
    >
      <div
        className="arrow"
      />
      Back to Top
    </a>
  </div>
  <div
    className="footer-logo"
  >
    <a
      href="/"
    />
  </div>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-3 col-xs-6"
      >
        <h1>
          Capeia
        </h1>
        <ul>
          <li>
            <a
              href="/welcome"
            >
              About the Platform
            </a>
          </li>
          <li>
            <a
              href="/faq"
            >
              Frequently Asked Questions
            </a>
          </li>
          <li>
            <a
              href="/info-for-institutes"
            >
              Information for Research Institutes
            </a>
          </li>
          <li>
            <a
              href="https://www.twitter.com/CapeiaScience"
              target="_blank"
            >
              <div
                className="twitterIcon"
              >
                <div
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                >
                  <svg
                    className="social-icon social-icon--twitter "
                    fill="#2f3240"
                    height={20}
                    viewBox="0 0 64 64"
                    width={20}
                  >
                    <g>
                      <circle
                        cx="32"
                        cy="32"
                        fill="#00aced"
                        r="31"
                        style={
                          Object {
                            "fill": "white",
                            "fillOpacity": 0.8,
                          }
                        }
                      />
                    </g>
                    <g>
                      <path
                        d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
                      />
                    </g>
                  </svg>
                </div>
              </div>
              Stay in Touch
            </a>
          </li>
        </ul>
      </div>
      <div
        className="col-sm-3 col-xs-6"
      >
        <h1>
          Ressorts
        </h1>
        <ul>
          <li>
            <a
              href="/stem-cells"
            >
              Stem Cells
            </a>
          </li>
          <li>
            <a
              href="/brain-and-nerves"
            >
              Brain & Nerves
            </a>
          </li>
          <li>
            <a
              href="/cancer"
            >
              Cancer
            </a>
          </li>
          <li>
            <a
              href="/zoology"
            >
              Zoology
            </a>
          </li>
          <li>
            <a
              href="/paleobiology"
            >
              Paleobiology
            </a>
          </li>
          <li>
            <a
              href="/planetary-science"
            >
              Planetary Science
            </a>
          </li>
        </ul>
      </div>
      <div
        className="col-sm-3 col-xs-6"
      >
        <h1>
          Imprint
        </h1>
        <ul>
          <li>
            <a
              href="/who"
            >
              Who
            </a>
          </li>
          <li>
            <a
              href="/terms-of-use"
            >
              Terms of Use
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="https://www.newswire.com/news/launch-of-the-science-platform-capeia-19265208"
              target="_blank"
            >
              Press
            </a>
          </li>
          <li>
            <a
              href="/rules-and-guidelines"
            >
              Rules & Guidelines
            </a>
          </li>
        </ul>
      </div>
      <div
        className="col-sm-3 col-xs-6"
      >
        <h1>
          Contribute
        </h1>
        <ul>
          <li>
            <a
              href="/register"
            >
              Become an Author
            </a>
          </li>
          <li>
             
          </li>
          <li>
             
          </li>
          <li>
             
          </li>
          <li>
             
          </li>
          <li>
            <a
              href="/a-gap-in-society"
            >
              A Gap in Society
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="netidee"
      >
        Part of Netidee call 10
      </div>
    </div>
  </div>
</div>
`;
